code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -228,7 +165,62 @@
 			N.adjustBruteLoss(-1)
 			N.adjustFireLoss(-1)
 
+/obj/item/reagent_containers/food/snacks/sliceable/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>Alt-click to put something small inside.</span>")
+
+/obj/item/reagent_containers/food/snacks/sliceable/AltClick(mob/user)
+	var/obj/item/I = user.get_active_hand()
+	if(!I)
+		return
+	if(I.w_class > WEIGHT_CLASS_SMALL)
+		to_chat(user, "<span class='warning'>You cannot fit [I] in [src]!</span>")
+		return
+	var/newweight = GetTotalContentsWeight() + I.GetTotalContentsWeight() + I.w_class
+	if(newweight > MAX_WEIGHT_CLASS)
+		// Nope, no bluespace slice food
+		to_chat(user, "<span class='warning'>You cannot fit [I] in [src]!</span>")
+		return
+	if(!iscarbon(user))
+		return
+	if(!user.drop_item())
+		to_chat(user, "<span class='warning'>You cannot slip [I] inside [src]!</span>")
+		return
+	to_chat(user, "<span class='warning'>You slip [I] inside [src].</span>")
+	user.unEquip(I)
+	I.dropped(user)
+	total_w_class += I.w_class
+	add_fingerprint(user)
+	I.forceMove(src)
+
+/obj/item/reagent_containers/food/snacks/sliceable/attackby(obj/item/I, mob/user, params)
+	if((slices_num <= 0 || !slices_num) || !slice_path)
+		return 0
 
+	var/inaccurate = 1
+	if(I.sharp)
+		if(istype(I, /obj/item/kitchen/knife) || istype(I, /obj/item/scalpel))
+			inaccurate = 0
+	else
+		return TRUE
+	if(!isturf(loc) || !(locate(/obj/structure/table) in loc) && \
+			!(locate(/obj/machinery/optable) in loc) && !(locate(/obj/item/storage/bag/tray) in loc))
+		to_chat(user, "<span class='warning'>You cannot slice [src] here! You need a table or at least a tray to do it.</span>")
+		return TRUE
+	var/slices_lost = 0
+	if(!inaccurate)
+		user.visible_message("<span class='notice'>[user] slices [src]!</span>",
+		 "<span class='notice'>You slice [src]!</span>")
+	else
+		user.visible_message("<span class='notice'>[user] crudely slices [src] with [I]!</span>",
+			"<span class='notice'>You crudely slice [src] with your [I]</span>!")
+		slices_lost = rand(1,min(1,round(slices_num/2)))
+	var/reagents_per_slice = reagents.total_volume/slices_num
+	for(var/i=1 to (slices_num-slices_lost))
+		var/obj/slice = new slice_path (loc)
+		reagents.trans_to(slice,reagents_per_slice)
+	qdel(src)
+	..()
```

Comment:
we always want to `return ..()`

Keep in mind, this should be exclusive (that is to say, if all the other conditions fail).